extensions-contrib/statsd-emitter/src/main/java/io/druid/emitter/statsd/StatsDEmitter.java

```diff
@@ -0,0 +1,124 @@
+/*
+ *  Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ *  or more contributor license agreements. See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership. Metamarkets licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License. You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.druid.emitter.statsd;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+import com.metamx.common.logger.Logger;
+import com.metamx.emitter.core.Emitter;
+import com.metamx.emitter.core.Event;
+import com.metamx.emitter.service.ServiceMetricEvent;
+import com.timgroup.statsd.NonBlockingStatsDClient;
+import com.timgroup.statsd.StatsDClient;
+import com.timgroup.statsd.StatsDClientErrorHandler;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ */
+public class StatsDEmitter implements Emitter
+{
+
+  private final static Logger log = new Logger(StatsDEmitter.class);
+  private final static String DRUID_METRIC_SEPARATOR = "\\/";
+  private final static String STATSD_SEPARATOR = ":|\\|";
+
+  private final StatsDClient statsd;
+  private final StatsDEmitterConfig config;
+  private final DimensionConverter converter;
+
+  public StatsDEmitter(StatsDEmitterConfig config, ObjectMapper mapper) {
+    this.config = config;
+    this.converter = new DimensionConverter(mapper, config.getDimensionMapPath());
+    statsd = new NonBlockingStatsDClient(
+        config.getPrefix(),
+        config.getHostname(),
+        config.getPort(),
+        new StatsDClientErrorHandler()
+        {
+          @Override
+          public void handle(Exception exception)
+          {
+            log.error(exception, "Error sending metric to StatsD.");
```

Comment:
if this will log every emit error, you will need to print it every thousand or so otherwise you will flood  the logs with the same exception.